src/main/java/org/mockito/internal/junit/UniversalTestListener.java

```diff
@@ -44,8 +47,9 @@ public void testFinished(TestFinishedEvent event) {
         }
     }
 
-    private static void reportUnusedStubs(TestFinishedEvent event, Collection<Object> mocks) {
-        if (event.getFailure() == null) {
+    private void reportUnusedStubs(TestFinishedEvent event, Collection<Object> mocks) {
+        if (event.getFailure() == null && !stubbingLookupListener.mismatchesReported) {
+            //If there is some other failure (or mismatches were detected) don't report another exception to avoid confusion
```

Comment:
this comment should be above (I understand because I did follow on this code and did the review, but in 2 years form now if someone stumble on this code)